src/unix/fs.c

```diff
@@ -611,10 +645,43 @@ static ssize_t uv__fs_sendfile(uv_fs_t* req) {
 
 
 static ssize_t uv__fs_utime(uv_fs_t* req) {
+#if defined(__APPLE__)
+  struct attrlist attr_list;
+  struct timespec times[3];
+  unsigned i = 0;
+
+  bzero(&attr_list, sizeof(attr_list));
+
+  attr_list.bitmapcount = ATTR_BIT_MAP_COUNT;
+
+  if (!isnan(req->btime)) {
+    attr_list.commonattr |= ATTR_CMN_CRTIME;
+
+    times[i++].tv_sec = req->btime;
+    times[i].tv_nsec = (unsigned long)(req->btime * 1000000) % 1000000 * 1000;
+  }
+
+  if (!isnan(req->atime)) {
+    attr_list.commonattr |= ATTR_CMN_ACCTIME;
+
+    times[i++].tv_sec = req->atime;
+    times[i].tv_nsec = (unsigned long)(req->atime * 1000000) % 1000000 * 1000;
+  }
+
+  if (!isnan(req->mtime)) {
+    attr_list.commonattr |= ATTR_CMN_MODTIME;
+
+    times[i++].tv_sec = req->mtime;
+    times[i].tv_nsec = (unsigned long)(req->mtime * 1000000) % 1000000 * 1000;
+  }
```

Comment:
I thought so as well but I saw a lot of the same elsewhere.  I guess this is slightly different because it's 15 lines vs. 8-10 elsewhere.